src/precice/SolverInterface.hpp

```diff
@@ -772,6 +772,83 @@ class SolverInterface {
       int     valueIndex,
       double &value) const;
 
+  /**
+   * @brief setBoundingBoxes Define a region of interest on a received mesh
+   *        (<use-mesh ... from="otherParticipant />") in order to receive
+   *        only a certain mesh region.
+   *
+   *        This function is required if you don't want to use the preCICE
+   *        own mappings, but rather want to use your own solver for data
+   *        mapping. As opposed to the usual preCICE mapping, only a single
+   *        mesh (from the other participant) is now involved in this
+   *        situation since an 'own' mesh defined by the participant itself
+   *        is not required any more. In order to partition the receiving
+   *        interface mesh, the solver needs to define the mesh region he
+   *        wants to work on and write data to.
+   *
+   *        The complete concept on the receiving participant looks as follows
+   * @code
+   *    std::vector<double> boundingBoxes(dim * 2 * nBoundingBoxes);
+   *    // fill boundingBoxes ...
+   *    const int otherMeshID = precice.getMeshID("ReceivedMeshname");
+   *    const int writeDataID = precice.getDataID("WriteDataName", otherMeshID);
+   *
+   *    // Define region of interest, where we could obtain direct write access
+   *    precice.setBoundingBoxes(otherMeshID, boundingBox.data(), nBoundingBoxes);
+   *
+   *    double dt = precice.initialize();
+   *    // Get the size of the filtered mesh within the bounding box
+   *    // (provided by the coupling participant)
+   *    const int otherMeshSize = precice.getMeshVertexSize(writeDataID);
+   *
+   *    // Allocate a vector for the vertices
+   *    std::vector<double> otherSolverMesh(otherMeshSize * dim);
+   *    precice.getMeshVerticesWithIDs(otherMeshID, otherMeshSize, ids.data(), otherSolverMesh.data());
+   *    // continue with time loop and write data directly to writeDataID
+   * @endcode
```

Comment:
> Do we use this "@code" environment already?

That's a doxygen command. Doxygen will render this part as a code-block in our API documentation. Since this feature will be experimental in the first step, we could host it here as well?!

Reply:
I would prefer to have it on the website. There, we already have all means to mark it experimental (like a big warning block).
And we can properly explain how everything is connected (API, configuration option). I will need such an explanation to not forget already :grin:  
Otherwise, I am bit afraid that we will end up with duplicated examples.

Reply:
I removed the code example, referred to the website and kept the 'conceptual' description in the `SolverInterface`